usaspending_api/disaster/v2/views/disaster_base.py

```diff
@@ -77,6 +77,12 @@ def filter_by_defc_closed_periods() -> Q:
     return q & Q(submission__reporting_period_start__gte=str(REPORTING_PERIOD_MIN_DATE))
 
 
+def when_non_zero_award_spending(query):
+    return query.annotate(
+        total_outlay=Sum("gross_outlay_amount_by_award_cpe"), total_obligation=Sum("transaction_obligated_amount")
```

Comment:
I think the same issue with `gross_outlay_amount_by_award_cpe` being a cumulative total up to the current period is a problem here as well

Reply:
It's unfortunately implicit (and accidental), but in this case it looks like we are protected by the all_closed_defc_periods filter, which uses the final_of_fy logic here: https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/disaster/v2/views/disaster_base.py#L57

All endpoints using this method also use that. Better design patterns and tests should be made to ensure this stays the case in the future.

Reply:
Discussed outside of Github PR comments